engines/docker/engine.go

```diff
@@ -0,0 +1,103 @@
+package dockerengine
+
+import (
+	"sync"
+
+	docker "github.com/fsouza/go-dockerclient"
+	schematypes "github.com/taskcluster/go-schematypes"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"github.com/taskcluster/taskcluster-worker/runtime/caching"
+)
+
+type engine struct {
+	engines.EngineBase
+	m              sync.Mutex
+	Environment    *runtime.Environment
+	client         *docker.Client
+	monitor        runtime.Monitor
+	maxConcurrency int
+	engineConfig   configType
```

Comment:
You don't need both `configType` and `maxConcurrency`...
I would just add a single property:
```go
   config configType
```
since you're already inside the `engine` struct, there is no need to  call it `engineConfig`